server/config/config.go

```diff
@@ -428,9 +402,150 @@ type FleetConfig struct {
 	GeoIP            GeoIPConfig
 	Prometheus       PrometheusConfig
 	Packaging        PackagingConfig
+	MDM              MDMConfig
 	MDMApple         MDMAppleConfig `yaml:"mdm_apple"`
 }
 
+type MDMConfig struct {
+	AppleAPNsCert      string `yaml:"apple_apns_cert"`
+	AppleAPNsCertBytes string `yaml:"apple_apns_cert_bytes"`
+	AppleAPNsKey       string `yaml:"apple_apns_key"`
+	AppleAPNsKeyBytes  string `yaml:"apple_apns_key_bytes"`
+	AppleSCEPCert      string `yaml:"apple_scep_cert"`
+	AppleSCEPCertBytes string `yaml:"apple_scep_cert_bytes"`
+	AppleSCEPKey       string `yaml:"apple_scep_key"`
+	AppleSCEPKeyBytes  string `yaml:"apple_scep_key_bytes"`
+
+	// the following fields hold the parsed, validated TLS certificate set the
+	// first time AppleAPNs or AppleSCEP is called, as well as the PEM-encoded
+	// bytes for the certificate and private key.
+	appleAPNs        *tls.Certificate
+	appleAPNsPEMCert []byte
+	appleAPNsPEMKey  []byte
+	appleSCEP        *tls.Certificate
+	appleSCEPPEMCert []byte
+	appleSCEPPEMKey  []byte
+}
+
+type x509KeyPairConfig struct {
+	certPath  string
+	certBytes []byte
+	keyPath   string
+	keyBytes  []byte
+}
+
+func (x *x509KeyPairConfig) IsSet() bool {
```

Comment:
nit: should this be unexported?

Reply:
It is! The type is unexported, so this is not accessible from other packages. It's just a convention that I've seen used and that made sense to me, to distinguish between the API that is intended to be used for an unexported type, vs private/implementation details (there is none in this case).

Reply:
ohhh, I completely missed that, thanks!